internal/backends/winit/glwindow.rs

```diff
@@ -134,7 +140,16 @@ impl<Renderer: WinitCompatibleRenderer + 'static> WinitWindow for GLWindow<Rende
             None => return, // caller bug, doesn't make sense to call draw() when not mapped
         };
 
+        self.pending_redraw.set(false);
         self.renderer.render(&window.canvas, self);
+
+        // If during rendering a new redraw_request() was issued (for example in a rendering notifier callback), then
+        // pretend that an animation is running, so that we return Poll from the event loop to ensure a repaint as
+        // soon as possible.
+        if self.pending_redraw.get() {
+            i_slint_core::animations::CURRENT_ANIMATION_DRIVER
```

Comment:
the event_loop.rs could read the pending_redraw flag dirrectly instead of using this hack. Don't you think it would be cleaner?

(Could also be a return value of the draw function)

Reply:
That's a good idea.

Reply:
I've done that now, but the resulting duplication is a tiny little bit ugly. But maybe that's still better than this hack via the animation driver.